models/standalone-price/src/transformers.ts

```diff
@@ -97,18 +87,27 @@ const transformers = {
       'channel',
       'tiers',
     ],
-    addFields: ({ fields }) => {
+    replaceFields: ({ fields }) => {
       const customerGroupRef = fields.customerGroup
-        ? createCustomerGroupReference(fields.customerGroup, 'graphql')
+        ? Reference.random()
+            .typeId('customer-group')
+            .id(fields.customerGroup.id)
+            .buildGraphql<TReferenceGraphql<'customer-group'>>()
         : null;
+
       const channelRef = fields.channel
-        ? KeyReference.random()
+        ? Reference.random()
             .typeId('channel')
-            .key(fields.channel.key)
+            .id(fields.channel.id)
```

Comment:
AFAIK in both the GraphQL schema and the `TStandalonePriceGraphql` within the repository, the `channelRef` and `customerRef` fields return `id` values not `key`s.

Reply:
from `packages-shared/types/src/generated/ctp.ts`:

```
export type TChannelReferenceIdentifier = {
  __typename?: 'ChannelReferenceIdentifier';
  id?: Maybe<Scalars['String']['output']>;
  key?: Maybe<Scalars['String']['output']>;
  typeId: Scalars['String']['output'];
};
```

```
export type TCustomerGroupReferenceIdentifier = {
  __typename?: 'CustomerGroupReferenceIdentifier';
  id?: Maybe<Scalars['String']['output']>;
  key?: Maybe<Scalars['String']['output']>;
  typeId: Scalars['String']['output'];
};
```

@nima-ct `key` is a valid field for graphql references of both `customerGroup` and `channel`

Reply:
but, youre right, `customerGroupRef` only allows for `id` in the graphql schema

Reply:
Hi @ByronDWall, I’m not sure if the types you mentioned apply to the StandalonePrice. In the ctp file, both channelRef and customerGroupRef are typed as Reference for their values.
```
export type TStandalonePrice = TVersioned & {
  __typename?: 'StandalonePrice';
  ...
  channelRef?: Maybe<TReference>;
  customerGroupRef?: Maybe<TReference>;
};
```
```
export type TReference = {
  __typename?: 'Reference';
  id: Scalars['String']['output'];
  typeId: Scalars['String']['output'];
};
```

Reply:
`TResourceIdentifierInput` allows a key and is used in `Mutation.CreateStandalonePrice`, but not in `Query.StandalonePrice`.

Your PR is correct, and I think this is where the original confusion originated from.